hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DiskBalancer.java

```diff
@@ -504,15 +503,13 @@ private void createWorkPlan(NodePlan plan) throws DiskBalancerException {
     Map<String, String> storageIDToVolBasePathMap = new HashMap<>();
     FsDatasetSpi.FsVolumeReferences references;
     try {
-      try(AutoCloseableLock lock = this.dataset.acquireDatasetReadLock()) {
```

Comment:
Just suggest to keep it and improve to #dataSetLock as other improvement. We should create another jira to remove it if this is not necessary now.

Reply:
this method just get volumes snapshot so no need to acquire dataset lock.